packages/react-native/ReactAndroid/src/test/java/com/facebook/react/animated/NativeAnimatedInterpolationTest.kt

```diff
@@ -0,0 +1,165 @@
+    assertThat(simpleInterpolation(0.0, input, output)).isEqualTo(100.0)
+    assertThat(simpleInterpolation(0.5, input, output)).isEqualTo(150.0)
+    assertThat(simpleInterpolation(0.8, input, output)).isEqualTo(180.0)
+    assertThat(simpleInterpolation(1.0, input, output)).isEqualTo(200.0)
+  }
+
+  @Test
+  fun testWiderInputRange() {
+    val input = doubleArrayOf(2000.0, 3000.0)
+    val output = doubleArrayOf(1.0, 2.0)
+    assertThat(simpleInterpolation(2000.0, input, output)).isEqualTo(1.0)
+    assertThat(simpleInterpolation(2250.0, input, output)).isEqualTo(1.25)
+    assertThat(simpleInterpolation(2800.0, input, output)).isEqualTo(1.8)
+    assertThat(simpleInterpolation(3000.0, input, output)).isEqualTo(2.0)
+  }
+
+  @Test
+  fun testManySegments() {
+    val input = doubleArrayOf(-1.0, 1.0, 5.0)
+    val output = doubleArrayOf(0.0, 10.0, 20.0)
+    assertThat(simpleInterpolation(-1.0, input, output)).isEqualTo(0.0)
+    assertThat(simpleInterpolation(0.0, input, output)).isEqualTo(5.0)
+    assertThat(simpleInterpolation(1.0, input, output)).isEqualTo(10.0)
+    assertThat(simpleInterpolation(2.0, input, output)).isEqualTo(12.5)
+    assertThat(simpleInterpolation(5.0, input, output)).isEqualTo(20.0)
+  }
+
+  @Test
+  fun testExtendExtrapolate() {
+    val input = doubleArrayOf(10.0, 20.0)
+    val output = doubleArrayOf(0.0, 1.0)
+    assertThat(simpleInterpolation(30.0, input, output)).isEqualTo(2.0)
+    assertThat(simpleInterpolation(5.0, input, output)).isEqualTo(-0.5)
+  }
+
+  @Test
+  fun testClampExtrapolate() {
+    val input = doubleArrayOf(10.0, 20.0)
+    val output = doubleArrayOf(0.0, 1.0)
+    assertThat(
+      InterpolationAnimatedNode.interpolate(
+        30.0,
+        input,
+        output,
+        InterpolationAnimatedNode.EXTRAPOLATE_TYPE_CLAMP,
+        InterpolationAnimatedNode.EXTRAPOLATE_TYPE_CLAMP
+      )
+    ).isEqualTo(1.0)
+    assertThat(
+      InterpolationAnimatedNode.interpolate(
+        5.0,
+        input,
+        output,
+        InterpolationAnimatedNode.EXTRAPOLATE_TYPE_CLAMP,
+        InterpolationAnimatedNode.EXTRAPOLATE_TYPE_CLAMP
+      )
+    ).isEqualTo(0.0)
+  }
+
+  @Test
+  fun testIdentityExtrapolate() {
+    val input = doubleArrayOf(10.0, 20.0)
+    val output = doubleArrayOf(0.0, 1.0)
+    assertThat(
+      InterpolationAnimatedNode.interpolate(
+        30.0,
+        input,
+        output,
+        InterpolationAnimatedNode.EXTRAPOLATE_TYPE_IDENTITY,
+        InterpolationAnimatedNode.EXTRAPOLATE_TYPE_IDENTITY
+      )
+    ).isEqualTo(30.0)
+    assertThat(
+      InterpolationAnimatedNode.interpolate(
+        5.0,
+        input,
+        output,
+        InterpolationAnimatedNode.EXTRAPOLATE_TYPE_IDENTITY,
+        InterpolationAnimatedNode.EXTRAPOLATE_TYPE_IDENTITY
+      )
+    ).isEqualTo(5.0)
+  }
+
+  @Test
+  fun testInterpolateColor() {
+    val input = doubleArrayOf(0.0, 1.0)
+    val output = intArrayOf(-0x1000000, -0x10000)
```

Comment:
Not a big fan of this conversion.
Can we use instead:

```
0xFF000000.toInt()
```

and make sure the original hex values are preserved? (valid for the whole test)